firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -153,6 +158,59 @@ uint8_t TCMD_get_suffix_tag_uint64(const char *str, const char *tag_name, uint64
     return 0;
 }
 
+/// @brief Searches for a `str` like `\@tag_name=xxxx`, and sets uint8_t array `xxxx` into `value_dest`.
+/// @param str The string (haystack) to search with for the tag.
+/// @param tag_name The tag to search for, including the '@' and '='.
+/// @param value_dest The destination for the value. `*value_dest` will be set.
+/// @return 0 if the tag was found successfully. >0 if the tag was not found, or there was an error.
+/// @note This function will return an error if the character after the string is not in: `#\@\0!`
+uint8_t TCMD_get_suffix_tag_hex_array(const char *str, const char *tag_name, uint8_t *value_dest) {
+    // Find the tag in the string
+    int16_t tag_index = GEN_get_index_of_substring_in_array(str, strlen(str), tag_name);
+    if (tag_index < 0) {
+        return 1;
+    }
+
+    // Find the start of the value, then do safety check.
+    uint16_t value_start_index = tag_index + strlen(tag_name);
+    if (value_start_index >= strlen(str)) {
+        return 2;
+    }
+
+    // Find the end of the value
+    uint16_t value_end_index = value_start_index;
+    while (TCMD_is_char_valid_telecommand_name_char(str[value_end_index])) {
+        value_end_index++;
+    }
+
+    // Check that there was at least one digit
+    if (value_end_index <= value_start_index) {
+        return 3;
+    }
+
+    // Check that the character after the number is any of '#', '@', '!', or '\0'
+    if (str[value_end_index] != '#' && str[value_end_index] != '@' && str[value_end_index] != '\0' && str[value_end_index] != '!') {
+        return 4;
+    }
+
+    // Copy the value into a buffer
+    char value_str[65]; // sha256 is 32 bytes long. String must be 64 chars long + null terminator
+    memset(value_str, 0, sizeof(value_str));
+    if ((uint32_t)(value_end_index - value_start_index) > sizeof(value_str)) {
+        // Failure: string too long
+        return 5;
+    }
+    strncpy(value_str, str + value_start_index, value_end_index - value_start_index);
+    strncpy(value_str + (value_end_index - value_start_index), "\0", 1); // Ensure string is null-terminated
+
+    // Convert the value to a hex array
+    if(TCMD_str_to_hex_array(value_str, strlen(value_str), value_dest) != 0) {
```

Comment:
Replace this function with the existing parse hex array function listed in a prior comment